pandas/tseries/tools.py

```diff
@@ -437,6 +437,10 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
     if arg is None:
         return None
 
+    check_arg = np.array(arg) if np.iterable(arg) else np.array([arg])
```

Comment:
do

```
if unit is None:
    ....
```

move the actual is_numeric_dtype / is_scalar check (same one that is in origin) to a function (that you call in both places)